### PR TITLE
Added Pagination, Sorting, and Swagger to the existing controllers

### PR DESCRIPTION
**Pull request by ynik404**

### CODE REVIEW

File: src/main/java/mate/academy/controller/BookController.java [hunk 1]
```diff
@@ -17,34 +20,42 @@
...
     @DeleteMapping("/{id}")
+    @Operation(summary = "Delete book", description = "Delete book")
     public void delete(@PathVariable Long id) {
         bookService.deleteById(id);
     }
 
     @GetMapping("/search")
+    @Operation(summary = "Get all books by parameters",
+            description = "Get a list of all available books by parameters")
     public List<BookDto> search(BookSearchParameters searchParametersDto) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/controller/BookController.java:59]:
```suggestion
    public List<BookDto> search(BookSearchParameters searchParametersDto, @ParameterObject @PageableDefault Pageable pageable) {
```

File: src/main/java/mate/academy/controller/BookController.java [hunk 2]
```diff
@@ -17,34 +20,42 @@
...
 @RestController
 @RequestMapping("/books")
 public class BookController {
     private final BookService bookService;
 
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    @Operation(summary = "Get all books", description = "Get a list of all available books")
+    public List<BookDto> getAll(Pageable pageable) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/controller/BookController.java:32]:
```suggestion
    public List<BookDto> getAll(@ParameterObject @PageableDefault Pageable pageable) {
```

File: src/main/java/mate/academy/controller/BookController.java [hunk 3]
```diff
@@ -17,34 +20,42 @@
...
+    @Operation(summary = "Delete book", description = "Delete book")
     public void delete(@PathVariable Long id) {
         bookService.deleteById(id);
     }
 
     @GetMapping("/search")
+    @Operation(summary = "Get all books by parameters",
+            description = "Get a list of all available books by parameters")
     public List<BookDto> search(BookSearchParameters searchParametersDto) {
         return bookService.search(searchParametersDto);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/controller/BookController.java:60]:
```suggestion
        return bookService.search(searchParametersDto, pageable);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Let's fix some comments

**Review by okuzan — APPROVED** [review]

**Pull request merged**